src/coreclr/dlls/mscoree/exports.cpp

```diff
@@ -135,23 +138,36 @@ static void ConvertConfigPropertiesToUnicode(
         propertyKeysW[propertyIndex] = StringToUnicode(propertyKeys[propertyIndex]);
         propertyValuesW[propertyIndex] = StringToUnicode(propertyValues[propertyIndex]);
 
-        if (strcmp(propertyKeys[propertyIndex], "BUNDLE_PROBE") == 0)
+        if (strcmp(propertyKeys[propertyIndex], HOST_PROPERTY_BUNDLE_PROBE) == 0)
         {
             // If this application is a single-file bundle, the bundle-probe callback
             // is passed in as the value of "BUNDLE_PROBE" property (encoded as a string).
-            *bundleProbe = (BundleProbeFn*)_wcstoui64(propertyValuesW[propertyIndex], nullptr, 0);
+            if (*bundleProbe == nullptr)
+                *bundleProbe = (BundleProbeFn*)_wcstoui64(propertyValuesW[propertyIndex], nullptr, 0);
         }
-        else if (strcmp(propertyKeys[propertyIndex], "PINVOKE_OVERRIDE") == 0)
+        else if (strcmp(propertyKeys[propertyIndex], HOST_PROPERTY_PINVOKE_OVERRIDE) == 0)
         {
             // If host provides a PInvoke override (typically in a single-file bundle),
             // the override callback is passed in as the value of "PINVOKE_OVERRIDE" property (encoded as a string).
-            *pinvokeOverride = (PInvokeOverrideFn*)_wcstoui64(propertyValuesW[propertyIndex], nullptr, 0);
+            if (*pinvokeOverride == nullptr)
```

Comment:
This makes the behavior order dependent. Meaning the first out of the PINVOKE_OVERRIDE / RUNTIME_CONTRACT will win and the other will be ignored.

I want to make sure that this is intentional.
Personally I would prefer to have a clear priority defined and not be order dependent, but it's probably not a big deal (it should basically never happen that we have both at the same time).

Reply:
My intention was to make RUNTIME_CONTRACT win. I'll update and add a comment.

Reply:
Wait, isn't the contract winning, since it sets pinvokeOverride as long as the contract has it, but this only sets it if it isn't set? (Definitely still needs a comment)

Reply:
You're right - I didn't read it correctly. It will win (which is good). (Agree that it needs a comment :wink:)